src/librustc/hir/def.rs

```diff
@@ -92,12 +92,12 @@ impl DefKind {
             DefKind::Static => "static",
             DefKind::Enum => "enum",
             DefKind::Variant => "variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Fn) => "tuple variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Const) => "unit variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Fictive) => "struct variant",
+            DefKind::Ctor(CtorOf::Variant, CtorKind::Fn) => "tuple variant constructor",
+            DefKind::Ctor(CtorOf::Variant, CtorKind::Const) => "unit variant constructor",
+            DefKind::Ctor(CtorOf::Variant, CtorKind::Fictive) => "struct variant constructor",
```

Comment:
`CtorKind::Fictive` is not a working constructor, just some stub to reserve struct variants in value namespace, every use of it is an error, so it's better to keep the old wording (which would be reported if the constructor didn't exist at all).